src/devices/Camera/Camera/ProcessRunner.cs

```diff
@@ -0,0 +1,234 @@
+    /// </summary>
+    public int? PId => _process?.Id;
+
+    /// <summary>
+    /// Dispose the active running process
+    /// </summary>
+    public void Dispose()
+    {
+        _cts.Cancel();
+        Kill();
+    }
+
+    /// <summary>
+    /// Kill the process
+    /// </summary>
+    public void Kill()
+    {
+        try
+        {
+            if (_process != null && !_process.HasExited)
+            {
+#if NETSTANDARD2_0
+                _process.Kill();
+#else
+                _process.Kill(true);
+#endif
+                if (!_process.HasExited)
+                {
+                    _process.WaitForExit(5000);
+                }
+
+                _process.Dispose();
+            }
+        }
+        catch (Exception)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Retrieves the full command line equivalent to the one run from this class.
+    /// The working directory is either ProcessSettings.WorkingDirectory or, if null,
+    /// the Environment.CurrentDirectory.
+    /// The working directory is important when referring to files from the command line.
+    /// </summary>
+    public string GetFullCommandLine(string[] arguments)
+    {
+        var argsString = string.Join(" ", arguments);
+        return $"{_processSettings.Filename} {argsString}";
+    }
+
+    /// <summary>
+    /// Execute the process with a number of arguments. The target
+    /// Stream receives the stdout of the process
+    /// </summary>
+    public Task ExecuteAsync(string[] arguments, Stream? target)
+    {
+        var argsString = string.Join(" ", arguments);
+        return ExecuteAsync(argsString, target);
+    }
+
+    /// <summary>
+    /// Execute the process with a number of arguments.
+    /// The target Stream receives the stdout of the process, if any.
+    /// If the process is not expected to return any output (for example when
+    /// the app directly writes one or more files), the stream can be null
+    /// </summary>
+    public async Task ExecuteAsync(string argsString, Stream? target)
+    {
+        var processStartInfo = BuildOptions(argsString);
+        try
+        {
+            _process = new Process();
+            _process.EnableRaisingEvents = true;
+            _process.StartInfo = processStartInfo;
+            _process.Start();
+
+            if (target == null)
+            {
+#if NETSTANDARD2_0
+                _process.WaitForExit();
+#else
+                await _process.WaitForExitAsync(_cts.Token);
+#endif
+                return;
+            }
+
+            if (_processSettings.CaptureStderrInsteadOfStdout)
+            {
+                await _process.StandardError.BaseStream.CopyToAsync(target, _processSettings.BufferSize, _cts.Token);
+            }
+            else
+            {
+                await _process.StandardOutput.BaseStream.CopyToAsync(target, _processSettings.BufferSize, _cts.Token);
+            }
+
+            _process.WaitForExit(1000);
```

Comment:
> is 1 second a standard time? Can this vary?

Hard to say, it depends on the process being run.
I will put it in the `ProcessSettings` so that the user can modify the default value (1 sec is arbitrary).